### ISSUE TITLE
High level overview of builder

### ISSUE THREAD

**Issue by Seanny123** [nengo/nengo#1072]
The documentation for Nengo 2.0 has gotten way better over the last two years, especially given how the Nengo GUI makes it to create examples. One thing that's left to explain is the Nengo builder and how it works from a high level (kind of like the documentation I made for the Nengo GUI).

- Is documentation of the builder a useful endeavour? If so, to what extent and how?
- Are specific parts more important to document than others? For example, Terry and #1061 mention that some of our builder components are beings used by back-ends frequently. Should they be made a priority?
- Should this only be attempted after certain changes, like #453 are completed?

**Comment by tbekolay** [comment]
I tried to do this a little bit in 2.1.0: http://pythonhosted.org/nengo/backend_api.html#the-build-process

I'm also currently in the process of implementing the incremental build stuff (splitting build into two steps, one that fills in the objects then one that builds the signals and ops) so that PR will be a good time to make something more approachable like a notebook :+1:

**Comment by hunse** [comment]
> I'm also currently in the process of implementing the incremental build stuff

Can we have a list somewhere of what people are working on? I was thinking about starting on this earlier, and now I'm glad I didn't.

Also, do we want to start on something new like that before e.g. #453 is merged? Can we maybe have a mini-sprint to try to get through most of the PRs in the queue right now? Not an in-person sprint, just a focused online effort. There are a lot of PRs that are way out of date and will only get harder to rebase the more we do and the more we leave them. I'm thinking also of the `dtype` PR, and a few other ones that touch the builder.

**Comment by tbekolay** [comment]
> Can we have a list somewhere of what people are working on?

Yeah, I don't know what a good solution for this is; I searched around for a while but didn't find anything great, but that was a while ago. Maybe that should be on the weekly agenda?

> Can we maybe have a mini-sprint to try to get through most of the PRs in the queue right now?

SGTM! Will probably have to be after the summer school, but it would be great to do this with a group chat open for a few days or so... added to the agenda (though I will away for the next dev meeting -- I'm available pretty much any time after the summer school though).

**Comment by jgosmann** [comment]
> Can we have a list somewhere of what people are working on?

People could create issues (if not already existent) and assign themselves to them? Not sure how well that would work, but there would at least be some sort of notification and it is possible to filter issues by the work in progress label or by whether they are assigned (`assignee:*`) and that would produce some sort of such a list.
